pandas/tests/plotting/test_groupby.py

```diff
@@ -10,10 +10,54 @@
 import numpy as np
 
 from pandas.tests.plotting.common import TestPlotBase
+import pytest
 
 
 @td.skip_if_no_mpl
 class TestDataFrameGroupByPlots(TestPlotBase):
+    @pytest.mark.parametrize(
+        'bins, equal_bins',
+        zip([5, None, np.linspace(-3, 3, 10)], [True, False])
+    )
+    def test_hist_bins_match(self, bins, equal_bins):
+        # GH-22222
+        N = 100
+        df = DataFrame(np.append(np.linspace(-3, 3, N),
+                                 np.linspace(-1, 1, N)),
+                       columns=['rand'])
+        df['group'] = [0] * N + [1] * N
+        g = df.groupby('group')['rand']
+        axes = g.hist(bins=bins, alpha=0.7, equal_bins=equal_bins)
+        ax = axes[0]
+
+        if bins is None:
```

Comment:
Just generally avoid this type of code in tests - you aren't asserting anything about the functionality of the code being tested but rather just adding separate logic here in the test case

Reply:
The test should be slightly different, depending on the type provided for `bins`. Do you recommend dropping `parametrize` decorator and writing separate tests for these instead?